plugins/builds/create.js

```diff
@@ -16,18 +18,49 @@ module.exports = (server) => ({
             scope: ['user']
         },
         handler: (request, reply) => {
-            const Build = new Model.Build(
+            const build = new Model.Build(
                 server.settings.app.datastore,
-                server.settings.app.executor
+                server.settings.app.executor,
+                options.password
             );
             const payload = {
                 jobId: request.payload.jobId,
                 apiUri: request.server.info.uri,
                 tokenGen: (buildId) =>
                     request.server.plugins.login.generateToken(buildId, ['build'])
             };
+            const username = request.auth.credentials.username;
 
-            Build.create(payload, (err, data) => {
+            // Check if user can create a build
+            async.waterfall([
+                // Get job
+                (next) => build.job.get(request.payload.jobId, next),
```

Comment:
It's the job model. Build has these models tied to it:
https://github.com/screwdriver-cd/models/blob/master/lib/build.js#L73-L75